### ISSUE TITLE
MS-4a: MMM Analysis Scripts — Validate & Harden

### ISSUE THREAD

**Issue by tkhongsap** [tkhongsap/rag-mmm-platform#83]
## Objective

Validate and harden the existing MMM statistical modeling scripts. These are standalone scripts that read `data/mmm/model_ready.csv` and output JSON results to stdout.

> **Parallel track**: This milestone can run **in parallel** with MS-2 and MS-3 on a different team member.

## Current State

All 4 analysis scripts already exist in `src/platform/api/mmm_scripts/`:
- `regression.py`, `roi_analysis.py`, `budget_optimizer.py`, `adstock_curves.py`
- `src/platform/api/mmm_summary.py` — builds KPIs from `model_ready.csv`
- Need validation, edge case fixes, and `__init__.py`

## Key Results

| # | Key Result | Measure |
|---|-----------|---------|
| 1 | Ridge regression produces valid output | `regression.py` outputs JSON with `r_squared > 0.5`, `coefficients` dict, `coefficient_magnitudes` (normalized importance ranking) |
| 2 | ROI analysis computes per-channel ROI | `roi_analysis.py` outputs JSON with all 11 channels, `roi` and `marginal_roi` per channel |
| 3 | Budget optimizer respects constraints | `budget_optimizer.py` output: each channel within ±30% of current, total budget unchanged |
| 4 | Adstock curves render decay and saturation | `adstock_curves.py` outputs per-channel `raw`, `adstocked`, `saturated` arrays |
| 5 | Dashboard summary returns KPIs | `mmm_summary.py` returns `total_spend`, `total_units`, `channel_breakdown`, `weekly_spend` |

> **Important**: Ridge regression (`sklearn.linear_model.Ridge`) cannot produce p-values. Use `coefficient_magnitudes` (normalized absolute coefficients) for channel importance ranking instead.

## Tasks

- [ ] **4a.1** Run each script standalone and validate JSON output: `regression.py` (r_squared > 0.5, coefficients dict, coefficient_magnitudes), `roi_analysis.py` (11 channels with ROI), `budget_optimizer.py` (constraints hold: each channel ±30%, total unchanged), `adstock_curves.py` (per-channel arrays)
- [ ] **4a.2** Verify `mmm_summary.py` returns all required fields: `total_spend`, `total_units`, `total_revenue`, `channel_breakdown`, `weekly_spend`, `weeks_of_data`
- [ ] **4a.3** Add `src/platform/api/mmm_scripts/__init__.py` if missing
- [ ] **4a.4** Fix any issues found (edge cases, missing error handling, output format inconsistencies)

## Deliverables

| File | Type | Est. Lines |
|------|------|-----------|
| `src/platform/api/mmm_scripts/__init__.py` | New | empty |
| `src/platform/api/mmm_scripts/regression.py` | Exists — validate | ~100 |
| `src/platform/api/mmm_scripts/roi_analysis.py` | Exists — validate | ~100 |
| `src/platform/api/mmm_scripts/budget_optimizer.py` | Exists — validate | ~100 |
| `src/platform/api/mmm_scripts/adstock_curves.py` | Exists — validate | ~80 |
| `src/platform/api/mmm_summary.py` | Exists — validate | ~100 |

## Verification

```bash
python src/platform/api/mmm_scripts/regression.py | python -m json.tool
python src/platform/api/mmm_scripts/roi_analysis.py | python -m json.tool
python src/platform/api/mmm_scripts/budget_optimizer.py | python -m json.tool
python src/platform/api/mmm_scripts/adstock_curves.py | python -m json.tool
python -c "
from src.platform.api.mmm_summary import build_mmm_summary
s = build_mmm_summary()
assert 'total_spend' in s and 'channel_breakdown' in s
print('Summary OK')
"
```

## Dependencies

- **MS-1** (#80) — `data/mmm/model_ready.csv` must exist (already generated by existing data pipeline).

---
📋 Reference: [`docs/blueprint/milestones.md`](docs/blueprint/milestones.md) · [`docs/blueprint/team-execution-plan.md`](docs/blueprint/team-execution-plan.md)
